library/common/extensions/key_value/platform/c_types.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include "library/common/types/c_types.h"
+
+// NOLINT(namespace-envoy)
+
+#ifdef __cplusplus
+extern "C" { // function pointers
+#endif
+
+/**
+ * Function signature for reading value from implementation.
+ */
+typedef envoy_data (*envoy_kv_store_read_f)(envoy_data key, const void* context);
```

Comment:
Is there a reason you went with a synchronous approach here? Is the idea that consumers would take care of moving this to a separate thread if they don't want to block on reading/writing?

Reply:
Well, for `read` asynchronous would introduce a whole new calling paradigm ("onRead"?). Honestly, I'd think an implementation should probably do caching/pre-caching/memoization if read performance is likely to be a bottleneck somewhere. For `write`/`remove `, yes, I think that's also better off as a feature of the actual provided platform implementation.

Reply:
ok. Another thing to consider is whether or not errors should be propagated between the platform and native layers.